File: baseLib/src/main/java/me/ycdev/android/lib/common/utils/GsonHelper.java
package me.ycdev.android.lib.common.utils;

import android.support.annotation.NonNull;

import com.google.gson.JsonObject;

@SuppressWarnings({"unused", "WeakerAccess"})
public class GsonHelper {
    public static String optString(@NonNull JsonObject json, @NonNull String key, String defValue) {
        if (json.has(key)) {
            return json.get(key).getAsString();
        }
        return defValue;
    }

    public static boolean optBoolean(@NonNull JsonObject json, @NonNull String key, boolean defValue) {
        if (json.has(key)) {
            return json.get(key).getAsBoolean();
        }
        return defValue;
    }

    public static int optInt(@NonNull JsonObject json, @NonNull String key, int defValue) {
        if (json.has(key)) {
            return json.get(key).getAsInt();
        }
        return defValue;
    }

    public static long optLong(@NonNull JsonObject json, @NonNull String key, long defValue) {
        if (json.has(key)) {
            return json.get(key).getAsLong();
        }
        return defValue;
    }

    public static float optFloat(@NonNull JsonObject json, @NonNull String key, float defValue) {
        if (json.has(key)) {
            return json.get(key).getAsFloat();
        }
        return defValue;
    }

    public static double optDouble(@NonNull JsonObject json, @NonNull String key, double defValue) {
        if (json.has(key)) {
            return json.get(key).getAsDouble();
        }
        return defValue;
    }
}

File: baseLib/src/main/java/me/ycdev/android/lib/common/utils/GsonHelper.kt
package me.ycdev.android.lib.common.utils

import com.google.gson.JsonObject

object GsonHelper {
    fun optString(
        json: JsonObject,
        key: String,
        defValue: String?
    ): String? = if (json.has(key)) {
        json.get(key).asString
    } else {
        defValue
    }

    fun optBoolean(
        json: JsonObject,
        key: String,
        defValue: Boolean
    ): Boolean = if (json.has(key)) {
        json.get(key).asBoolean
    } else {
        defValue
    }

    fun optInt(
        json: JsonObject,
        key: String,
        defValue: Int
    ): Int = if (json.has(key)) {
        json.get(key).asInt
    } else {
        defValue
    }

    fun optLong(
        json: JsonObject,
        key: String,
        defValue: Long
    ): Long = if (json.has(key)) {
        json.get(key).asLong
    } else {
        defValue
    }

    fun optFloat(
        json: JsonObject,
        key: String,
        defValue: Float
    ): Float = if (json.has(key)) {
        json.get(key).asFloat
    } else {
        defValue
    }

    fun optDouble(
        json: JsonObject,
        key: String,
        defValue: Double
    ): Double = if (json.has(key)) {
        json.get(key).asDouble
    } else {
        defValue
    }
}
